test/luatest_helpers/server.lua

```diff
@@ -190,4 +190,28 @@ function Server:start(opts)
     getmetatable(getmetatable(self)).start(self, opts)
 end
 
+-- Eval a string similar to `Server:eval()` and reconnect to the server
+-- in case the connection has been dropped.
+function Server:eval_reconnect(code, ...)
```

Comment:
Is it really a general purpose function?

A function for retrying a request with an on demand reconnection seems meaningful (but requires request_timeout, time_budget, retry_count and so on). But the reconnect without retrying looks quite specific.

Maybe it is better to include it into a test.